src/metabase/util/date_2.clj

```diff
@@ -592,3 +592,24 @@
 (defmethod print-method Duration
   [d writer]
   (print-method (list 't/duration (str d)) writer))
+
+(defn temporal-str->iso8601-str
+  "Convert temporal string to iso8601 datetime without millis.
+
+  We store datetime values without millis in sqlite. That's not the case for other dbs. Also, some columns are stored
+  as date in sqlite, while other dbs use datetime types. This function makes it easy to share expected results between
+  sqlite and other dbs.
+
+  Use of this function for anything else is highly discouraged."
+  [tstr]
+  (when tstr
+    (let [t (parse tstr)
+          inst (cond (instance? LocalDate t)
+                     (.toInstant ^LocalDateTime (.atStartOfDay ^LocalDate t) java.time.ZoneOffset/UTC)
+
+                     (instance? LocalDateTime t)
+                     (.toInstant ^LocalDateTime t java.time.ZoneOffset/UTC)
+
+                     :else
+                     t)]
+      (format "yyyy-MM-dd'T'HH:mm:ss'Z'" inst))))
```

Comment:
Currently we store sqlite test data with second precision. Ideal solution would be to find a way of increasing it. That's over the scope of this PR.

Reply:
I understand SQLite can understand ISO8601 strings in "YYYY-MM-DD HH:MM:SS.SSS" format, millisecond precision should be achievable. See also https://www.sqlite.org/lang_datefunc.html#subsec.

Reply:
I've opened https://github.com/metabase/metabase/issues/45790 to track this.